### PR TITLE
Add fast getri_strided_batch

### PR DESCRIPTION
**Pull request by cfranken**
Add getri_strided_batch (similar to getrf_strided_batch). Allocation free inside, easier to use getrf_strided_batch and getri_strided_batch for fast batched matrix inversion now

### CODE REVIEW

**Comment by vchuravy** [comment]
Thanks @cfranken, can you add a couple of tests https://github.com/JuliaGPU/CUDA.jl/blob/master/test/cublas.jl

File: lib/cublas/wrappers.jl [hunk 1]
```diff
@@ -1489,6 +1489,71 @@ for (fname, elty) in
...
+    # cublasStatus_t cublasDgetriBatched(
+    #   cublasHandle_t handle, int n, double **A,
+    #   int lda, int *PivotArray, double **C,
+    #   int ldc, int *info, int batchSize)
+        function getri_batched!(n, Aptrs::CuVector{CuPtr{$elty}},
+                          lda, Cptrs::CuVector{CuPtr{$elty}},ldc,
+                          pivotArray::CuArray{Cint})
+            batchSize = length(Aptrs)
+            info = CuArray{Cint}(undef, batchSize)
+            CUBLAS.$fname(CUBLAS.handle(), n, Aptrs, lda, pivotArray, Cptrs, ldc, info, batchSize)
```
Comment (maleadt) [inline lib/cublas/wrappers.jl]: This is inside the CUBLAS module already, so you shouldn't need these prefixes.

File: lib/cublas/wrappers.jl [hunk 2]
```diff
@@ -1489,6 +1489,71 @@ for (fname, elty) in
...
+        #   cublasHandle_t handle, int n, double **A,
+        #   int lda, int *PivotArray, double **C,
+        #   int ldc, int *info, int batchSize)
+        function getri_batched!(A::Vector{<:CuMatrix{$elty}},
+                              C::Vector{<:CuMatrix{$elty}},
+                              pivotArray::CuMatrix{Cint})
+            n = size(A[1])[1]
+            lda = max(1, stride(A[1], 2))
+            ldc = max(1, stride(C[1], 2))
+            Aptrs = CUBLAS.unsafe_batch(A)
```
Comment (maleadt) [inline lib/cublas/wrappers.jl]: Same here

**Review by maleadt — CHANGES_REQUESTED** [review]

File: lib/cublas/wrappers.jl (hunk 1 above)
Comment (cfranken) [reply lib/cublas/wrappers.jl]: not sure I fully understand. Only a 
`getri_batched(A::Vector{<:CuMatrix{$elty}},pivotArray::CuMatrix{Cint})`
function existed, which allocated a new `C`. Or did I misunderstand (sorry, my first PR for Julia!)?

**Review by cfranken — COMMENTED** [review]
Removed CUBLAS prefixes and some commented lines

File: test/cublas.jl [hunk 3]
```diff
@@ -1598,11 +1598,16 @@ end
         @testset "getrf_strided_batched" begin
             local k
             # generate strided matrix
-            A = rand(elty,m,m,10)
+            A = rand(elty,m,m,10);
```
Comment (maleadt) [inline test/cublas.jl]: No need for these changes.

File: test/cublas.jl [hunk 4]
```diff
@@ -1598,11 +1598,16 @@ end
...
             # move to device
-            d_A = CuArray(A)
-            pivot, info, d_B = CUBLAS.getrf_strided_batched(d_A, false)
+            d_A = CuArray(A);
```
Comment (maleadt) [inline test/cublas.jl]: Same.

**Review by maleadt — COMMENTED** [review]
Thanks! Some minor nits.

File: test/cublas.jl (hunk 4 above)
Comment (cfranken) [reply test/cublas.jl]: done

File: test/cublas.jl (hunk 3 above)
Comment (cfranken) [reply test/cublas.jl]: done

**Review by maleadt — APPROVED** [review]

**Pull request merged**